test/test.refresh.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2013 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var assert = require('assert');
+var googleAuth = require('../lib/auth/googleauth.js');
+var nock = require('nock');
+var fs = require('fs');
+
+nock.disableNetConnect();
+
+// Creates a standard JSON credentials object for testing.
+function createJSON() {
+  return {
+    'client_secret': 'privatekey',
+    'client_id': 'client123',
+    'refresh_token': 'refreshtoken',
+    'type': 'authorized_user'
+  };
+}
+
+describe('Refresh Token auth client', function() {
+
+});
+
+describe('.fromJson', function () {
+
+  it('should error on null json', function () {
```

Comment:
For this and all other negative tests, use the flavor which provides a 'done' handler, and call done from within the callback. Currently, if the callback is never executed, the tests will pass.